### PR TITLE
feat: search variable service and controller 

### PR DESCRIPTION
**Pull request by nathansandi**
Implementation of Search Variable over `operate-variables` index as discussed with exporter team
- Service layer - Update the existent Variable Service developed in the past 
- Controller - Include Controller
- Unit Tests - Include unit tests

## Related issues

https://github.com/camunda/camunda/issues/22629

### CODE REVIEW

File: search/search-domain/src/main/java/io/camunda/search/entities/VariableEntity.java [hunk 1]
```diff
@@ -17,7 +17,5 @@ public final record VariableEntity(
     String fullValue,
     boolean isPreview,
     Long scopeKey,
-    Long processInstanceKey,
-    Long processDefinitionKey,
-    String bpmnProcessId,
+    Long processInstanceId,
```
Comment (ralfpuchert) [inline search/search-domain/src/main/java/io/camunda/search/entities/VariableEntity.java:20]: Shouldn't it be `processInstanceKey`? See example:

image

File: search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/filter/VariableFilterTransformer.java [hunk 2]
```diff
@@ -35,44 +35,55 @@ public VariableFilterTransformer(
...
-    return SearchQueryBuilders.term().field("value").value(typedValue).build().toSearchQuery();
-  }
-
   private SearchQuery getScopeKeyQuery(final List<Long> scopeKey) {
     return longTerms("scopeKey", scopeKey);
   }
 
   private SearchQuery getProcessInstanceKeyQuery(final List<Long> processInstanceKey) {
-    return longTerms("processInstanceKey", processInstanceKey);
+    return longTerms("processInstanceId", processInstanceKey);
```
Comment (ralfpuchert) [inline search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/filter/VariableFilterTransformer.java:75]: Shouldn't it be `processInstanceKey`. See also example object below.

File: search/search-client-query-transformer/src/test/java/io/camunda/search/clients/transformers/filter/VariableQueryTransformerTest.java [hunk 3]
```diff
@@ -41,131 +36,125 @@ public void shouldQueryOnlyByVariables() {
...
+    final var queryVariant = searchRequest.queryOption();
+    assertThat(queryVariant)
+        .isInstanceOfSatisfying(
+            SearchBoolQuery.class,
+            (t) -> {
+              assertThat(t.must().get(0).queryOption())
+                  .isInstanceOfSatisfying(
+                      SearchTermQuery.class,
+                      (term) -> {
+                        assertThat(term.field()).isEqualTo("processInstanceId");
```
Comment (ralfpuchert) [inline search/search-client-query-transformer/src/test/java/io/camunda/search/clients/transformers/filter/VariableQueryTransformerTest.java:87]: Same here.

File: search/search-client-query-transformer/src/test/java/io/camunda/search/clients/transformers/sort/VariableSortTest.java [hunk 4]
```diff
@@ -42,4 +42,135 @@ public void shouldApplySortConditionByValueASC() {
...
+
+    // then
+    Assertions.assertThat(sort).isNotNull();
+    Assertions.assertThat(sort).hasSize(2);
+
+    final boolean sortByProcessInstanceKeyConditionCheck =
+        sort.stream()
+            .anyMatch(
+                s ->
+                    s.field().field().equals("processInstanceId")
```
Comment (ralfpuchert) [inline search/search-client-query-transformer/src/test/java/io/camunda/search/clients/transformers/sort/VariableSortTest.java:171]: Same here.

File: search/search-domain/src/main/java/io/camunda/search/sort/VariableSort.java [hunk 5]
```diff
@@ -30,6 +30,31 @@ public Builder value() {
...
+      return this;
+    }
+
+    public Builder scopeKey() {
+      currentOrdering = new FieldSorting("scopeKey", null);
+      return this;
+    }
+
+    public Builder processInstanceKey() {
+      currentOrdering = new FieldSorting("processInstanceId", null);
```
Comment (ralfpuchert) [inline search/search-domain/src/main/java/io/camunda/search/sort/VariableSort.java:54]: Same here.

**Review by ralfpuchert — CHANGES_REQUESTED** [review]
Thank you! Looks good so far. You replaced `processInstanceKey` with `processInstanceId` for `VariableEntity` and I can't see process instance key in search results. Do I miss anything?

image

**Comment by nathansandi** [comment]
Thanks @ralfpuchert, I am provided the corrections,
I misunderstood with the value on tasklist-list-view that uses processInstanceId 
I am providing the fix :)

**Review by ralfpuchert — APPROVED** [review]
Nice! Thanks. Now we can see process instance key:
image

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 6]
```diff
@@ -2434,6 +2477,68 @@ components:
...
+        scopeKey:
+          type: integer
+          format: int64
+        processInstanceKey:
+          type: integer
+          format: int64
+        tenantId:
+          type: string
+        isTruncated:
+          type: boolean
```
Comment (pepopowitz) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:2508]: :x: Can you please add descriptions to these fields?

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 7]
```diff
@@ -2434,6 +2477,68 @@ components:
...
```
Comment (pepopowitz) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:2541]: :x: Can you please add descriptions to these fields?

**Review by pepopowitz — APPROVED** [review]
Please add descriptions!

**Pull request merged**